sphinx_immaterial/apidoc/python/autodoc_property_type.py

```diff
@@ -16,6 +16,11 @@
 else:
     stringify_annotation = sphinx.util.typing.stringify  # type: ignore[attr-defined]
 
+if sphinx.version_info >= (7, 3):
+    from sphinx.domains.python._annotations import _parse_annotation  # type: ignore[import-not-found]  # pylint: disable=import-error,no-name-in-module
+else:
+    from sphinx.domains.python import _parse_annotation
+
```

Comment:
Hmm --- I realize that I steered you wrong on this --- since we monkey patch this, we can run into problems by importing it.  Instead I guess we should just create an alias for the module.  We will also need to verify that our monkey patches still work --- e.g. if something in the Python domain itself imports this symbol, then we have to replace the imported symbol as well as the original one.

Reply:
Lately, I've been wondering if it would be easier to have our own "better domains" that somehow share/merge data with the vanilla domains (where applicable).

Reply:
Importing as alias seemed to be an over-complication; some of our modules do need to use other API of the python domain. Also, mypy doesn't like it when you import a module and then import it again under an alias (even if only conditionally imported as an alias).

I adjusted the unit test to assert that all modules in `sphinx.domains.python` which import `_parse_annotations()` are consistently monkeypatched.